Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx

```diff
@@ -694,6 +694,25 @@ int vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation(vtkMRMLSegm
   }
 
   // Read succeeded
+  int ret = vtkOrientedImageDataResample::IsSegmentationScalarTypeValid(imageData);
+  switch (ret)
+  {
+    case vtkOrientedImageDataResample::TYPE_CONVERSION_TRUNCATION_NEEDED:
+      vtkWarningToMessageCollectionMacro(this->GetUserMessages(), "vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation",
+        "Segmentation is a floating point scalar type and will be cast to an integer type. Voxel values may be truncated");
+      break;
+    case vtkOrientedImageDataResample::TYPE_CONVERSION_CLAMPING_NEEDED:
+       vtkWarningToMessageCollectionMacro(this->GetUserMessages(), "vtkMRMLSegmentationStorageNode::ReadBinaryLabelmapRepresentation",
```

Comment:
It would be nice to able to return this as userMessages to allow the error message to be displayed in the GUI.

Reply:
Isn't this one returned already?